### ISSUE TITLE
Module 13 - warning and error on creating the AD application

### ISSUE THREAD

**Issue by Floriszz** [MicrosoftLearning/AZ-304-Microsoft-Azure-Architect-Design#97]
# Module: 13
#### Step: 2

Description of issue
running the scripts gives the following feedback in the command prompt(Cloud CLI)
**WARNING: Upcoming breaking changes in the cmdlet 'New-AzADApplication' :

- The output type is changing from the existing type :'Microsoft.Azure.Commands.ActiveDirectory.PSADApplication' to the new type :'Microsoft.Azure.PowerShell.Cmdlets.Resources.MSGraph.Models.ApiV10.IMicrosoftGraphApplication'

- The parameter : 'Password' is changing.
- Change description : Parameter Password will not be supported, server generated secret text could be found in response
Note : Go to https://aka.ms/azps-changewarnings for steps to suppress this breaking change warning, and other information on breaking changes in Azure PowerShell.**

**_New-AzADApplication: Values of identifierUris property must use a verified domain of the organization or its subdomain: 'http://az30304aadsp'_**

Repro steps:

1. Run the script:
$password = 'Pa55w.rd1234.@z304'
$securePassword = ConvertTo-SecureString -Force -AsPlainText -String $password
$az30304aadapp = New-AzADApplication -DisplayName 'az30304aadsp' -HomePage 'http://az30304aadsp' -IdentifierUris 'http://az30304aadsp' -Password $securePassword

1.
1.

**Comment by Floriszz** [comment]
In order to have the script work, I added the fqdn from my user account from Azure Acitive Directory the the '-HomePage' and '-IdentifierUris' parameters.
So instead of http://az30304aadsp I made it; 'http://az30304aadsp.{myaccountname}.onmicrosoft.com', where {myaccountname} is my unique part in the fqdn part after the @ and 'before .onmicrosoft.com' in Azure AD.

This solves the last error to be able to get the app registration done. It does not solve the warnings.

**Comment by cdanvergara** [comment]
Tried the same and can confirm that works with this workaround!

**Comment by ElaMark** [comment]
Do this, substituting your own domain name:

$az30304aadapp = New-AzADApplication -DisplayName 'az30304aadsp' -HomePage 'http://your-aad-domain-name' IdentifierUris 'http://your-aad-domain-name' -SignInAudience AzureADMyOrg 

  

New-AzADServicePrincipal -ApplicationId $az30304aadapp.AppID 


I'll raise a PR

**Comment by Usamawahabkhan** [comment]
still having issue with password parameter because API has been changed as per doc

**Comment by ElaMark** [comment]
Yep, sure. Don't use a password. Generate an new token secret using registered applications in AAD. Use that instead
